objective.py

```diff
@@ -6,54 +6,130 @@
+            ('inter_session', 3, None, 'session_E')
 
+             ],
+    }
+    # The solvers will train on all the subject except subject_test.
+    # It will be the same for the sessions.
     # Minimal version of benchopt required to run this benchmark.
     # Bump it up if the benchmark depends on a new feature of benchopt.
     min_benchopt_version = "1.3.2"
 
-    def set_data(self, X, y):
+    def set_data(self, dataset, paradigm_name, sfreq):
         # The keyword arguments of this function are the keys of the dictionary
         # returned by `Dataset.get_data`. This defines the benchmark's
         # API to pass data. This is customizable for each benchmark.
+        # The dictionary defines the keyword arguments
+        # for `Objective.set_data`
+
+        data_split_subject = dataset.split('subject')
+
+        if self.evaluation_process == 'intra_subject':
+
+            dataset = data_split_subject[str(self.subject)]
+            X = SliceDataset(dataset, idx=0)
+            y = np.array([y for y in SliceDataset(dataset, idx=1)])
+
+            # maybe we need to do here different process for each subjects
 
-        X_train, X_test, y_train, y_test = train_test_split(X, y)
-        self.X_train, self.y_train = X_train, y_train
-        self.X_test, self.y_test = X_test, y_test
+            X_train, X_test, y_train, y_test = train_test_split(X, y)
+            self.X_train, self.y_train = X_train, y_train
+            self.X_test, self.y_test = X_test, y_test
+
+        elif self.evaluation_process == 'inter_subject':
+
+            sujet_test = self.subject_test
+            data_subject_test = data_split_subject[str(sujet_test)]
+            n_subject = len(data_split_subject)
+            data_subject_train = []
+            for i in range(1, n_subject+1):
+                if i != sujet_test:
+                    data_subject_train += data_split_subject[str(i)]
+
+            X_test = SliceDataset(data_subject_test, idx=0)
+            y_test = np.array([y for y in SliceDataset(data_subject_test,
+                                                       idx=1)])
+
+            X_train = SliceDataset(data_subject_train, idx=0)
+            y_train = np.array([y for y in SliceDataset(data_subject_train,
+                                                        idx=1)])
+
+            self.X_train, self.y_train = X_train,  y_train
+            self.X_test, self.y_test = X_test, y_test
+
+        elif self.evaluation_process == 'inter_session':
+
+            data_subject = data_split_subject[str(self.subject)]
+            data_split_session = data_subject.split('session')
+            session_test = self.session_test
+            data_session_test = data_split_session[session_test]
+            data_session_train = []
+            for clé in data_split_session.items():
+                if clé[0] != str(session_test):
+                    data_session_train += data_split_session[clé[0]]
```

Comment:
:rofl: you killed me here
Avoid french in the code, and in particular accents
```suggestion
            for key in data_split_session.items():
                if key[0] != str(session_test):
                    data_session_train += data_split_session[key[0]]
```